sim/src/index/componentType.h

```diff
@@ -0,0 +1,62 @@
+#ifndef BENTOBOX_COMPONENTTYPEINDEX_H
+#define BENTOBOX_COMPONENTTYPEINDEX_H
+
+#include <unordered_map>
+#include <functional>
+#include <core/ics/component.h>
+#include <core/ics/componentSet.h>
+
+namespace ics::index {
+    class ComponentType {
+    private:
+        std::unordered_map<std::type_index, size_t> map;
+        size_t compTypeIndex = 0;
```

Comment:
Index might be too overloaded at this point. We have a concept `Index` in ICS and seeing it here might trigger that concept instead of what this variable is.


I see that you refer to this as `group` elsewhere so maybe that would be a better name.

Reply:
Resolved in 0dba137